link.go

```diff
@@ -334,9 +367,10 @@ Loop:
 					// decrement link-credit after entire message transferred
 					if !tr.More {
 						l.deliveryCount++
-						l.linkCredit--
```

Comment:
If we don't decrement, how do we keep track of credits?

Reply:
We're still decrementing. sync.AddUint32 modifies it's first argument.

https://pkg.go.dev/sync/atomic#AddUint32